src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -290,19 +289,33 @@ public IAsyncCursor<TResult> Execute(IReadBinding binding, CancellationToken can
             }
 
             IAsyncCursor<RawBsonDocument> cursor;
+            IBatchInfo batchInfo;
+            BsonTimestamp initialOperationTime;
             using (var channelSource = binding.GetReadChannelSource(cancellationToken))
             using (var channel = channelSource.GetChannel(cancellationToken))
             using (var channelBinding = new ChannelReadBinding(channelSource.Server, channel, binding.ReadPreference, binding.Session.Fork()))
             {
-                cursor = ExecuteAggregateOperation(channelBinding, resuming: false, cancellationToken);
-                SaveInitialOperationTimeIfRequired(channel, channelBinding, cursor);
+                cursor = ExecuteAggregateOperation(channelBinding, cancellationToken);
+                batchInfo = GetBatchInfoOrThrow(cursor);
```

Comment:
No need for a helper method, just use:

    cursorBatchInfo = (ICursorBatchInfo)cursor;

Reply:
Done